devices/rp/rp2040.xml

```diff
@@ -0,0 +1,423 @@
+      <vector position="12" name="DMA_IRQ_1"/>
+      <vector position="13" name="IO_IRQ_BANK0"/>
+      <vector position="14" name="IO_IRQ_QSPI"/>
+      <vector position="15" name="SIO_IRQ_PROC0"/>
+      <vector position="16" name="SIO_IRQ_PROC1"/>
+      <vector position="17" name="CLOCKS_IRQ"/>
+      <vector position="18" name="SPI0_IRQ"/>
+      <vector position="19" name="SPI1_IRQ"/>
+      <vector position="20" name="UART0_IRQ"/>
+      <vector position="21" name="UART1_IRQ"/>
+      <vector position="22" name="ADC_IRQ_FIFO"/>
+      <vector position="23" name="I2C0_IRQ"/>
+      <vector position="24" name="I2C1_IRQ"/>
+      <vector position="25" name="RTC_IRQ"/>
+      <memory name="flash" access="rx" start="0x10000000" size="2097152"/>
+      <memory name="ram" access="rwx" start="0x20000000" size="262144"/>
+      <memory name="scratch_x" access="rwx" start="0x20040000" size="4096"/>
+      <memory name="scratch_y" access="rwx" start="0x20041000" size="4096"/>
+    </driver>
+    <driver name="adc" type="rp"/>
+    <driver name="busctrl" type="rp"/>
+    <driver name="clocks" type="rp"/>
+    <driver name="dma" type="rp"/>
+    <driver name="i2c" type="rp">
+      <instance value="0"/>
+      <instance value="1"/>
+    </driver>
+    <driver name="pio" type="rp">
+      <instance value="0"/>
+      <instance value="1"/>
+    </driver>
+    <driver name="ppb" type="rp"/>
+    <driver name="psm" type="rp"/>
+    <driver name="pwm" type="rp"/>
+    <driver name="resets" type="rp"/>
+    <driver name="rosc" type="rp"/>
+    <driver name="rtc" type="rp"/>
+    <driver name="sio" type="rp"/>
+    <driver name="spi" type="rp">
+      <instance value="0"/>
+      <instance value="1"/>
+    </driver>
+    <driver name="syscfg" type="rp"/>
+    <driver name="sysinfo" type="rp"/>
+    <driver name="tbman" type="rp"/>
+    <driver name="timer" type="rp"/>
+    <driver name="uart" type="rp">
+      <instance value="0"/>
+      <instance value="1"/>
+    </driver>
+    <driver name="usbctrl_dpram" type="rp"/>
+    <driver name="usbctrl_regs" type="rp"/>
+    <driver name="vreg_and_chip_reset" type="rp"/>
+    <driver name="watchdog" type="rp"/>
+    <driver name="xip_ctrl" type="rp"/>
+    <driver name="xip_ssi" type="rp"/>
+    <driver name="gpio" type="rp">
+      <gpio port="bank0" pin="0" name="0">
+        <func name="jtag_tck" value="0"/>
+        <func name="spi0_rx" value="1"/>
+        <func name="uart0_tx" value="2"/>
+        <func name="i2c0_sda" value="3"/>
+        <func name="pwm_a_0" value="4"/>
+        <func name="sio_0" value="5"/>
+        <func name="pio0_0" value="6"/>
+        <func name="pio1_0" value="7"/>
+        <func name="usb_muxing_overcurr_detect" value="9"/>
+      </gpio>
+      <gpio port="bank0" pin="1" name="1">
+        <func name="jtag_tms" value="0"/>
+        <func name="spi0_ss_n" value="1"/>
+        <func name="uart0_rx" value="2"/>
+        <func name="i2c0_scl" value="3"/>
```

Comment:
Should be split this up like in STM32? `driver="i2c" instance="0" name="scl" value="3"`
Could be useful for the connect API.